pkg/mount/block/blockdev_linux.go

```diff
@@ -556,12 +556,60 @@ func (b BlockDevices) FilterPartLabel(label string) BlockDevices {
 	return b.FilterNames(names...)
 }
 
+// FilterAllowPCIString parses a string in the format vendor:device,vendor:device
+// and returns a list of BlockDev objects whose backing pci devices match
+// the vendor:device pairs passed in. All values are treated as hex.
+// E.g. 0x8086:0xABCD,8086:0x1234
+func (b BlockDevices) FilterAllowPCIString(allowlist string) (BlockDevices, error) {
+	pciList, err := parsePCIList(allowlist)
+	if err != nil {
+		return nil, err
+	}
+	return b.FilterAllowPCI(pciList), nil
+}
+
+// FilterAllowPCI returns a list of BlockDev objects whose backing
+// pci devices match the allowlist of PCI devices passed in.
+// FilterAllowPCI discards entries which don't have a matching PCI vendor
+// and device ID as an entry in the allowlist.
+func (b BlockDevices) FilterAllowPCI(allowlist pci.Devices) BlockDevices {
+	type mapKey struct {
+		vendor, device uint16
+	}
+	m := make(map[mapKey]bool)
+
+	for _, v := range allowlist {
+		m[mapKey{v.Vendor, v.Device}] = true
+	}
+	Debug("allow map is %v", m)
+
+	partitions := make(BlockDevices, 0)
+	for _, device := range b {
+		p, err := device.PCIInfo()
+		if err != nil {
+			// In the case of an error, we err on the safe side and choose not to block it.
+			// Not all block devices are backed by a pci device, for example SATA drives.
+			Debug("Failed to find PCI info; %v", err)
+			partitions = append(partitions, device)
+			continue
+		}
+		if _, ok := m[mapKey{p.Vendor, p.Device}]; ok {
```

Comment:
can you consider remove the else branch here by favoring sth like

if _, ok := m[mapKey{p.Vendor, p.Device}];  !ok {
    log.Printf("Blocking device %v since it doesn't appear in allowlist", device.Name)
    continue
}
// Included in allowlist, we're good to go
Debug("Not blocking device %v, with pci %v, in map", device, p)
partitions = append(partitions, device)

Reply:
Done.